### PR TITLE
Generator rgb v2

### PR DESCRIPTION
**Pull request by bdilly**
v1 changelog:
  * use json_loop macro on generator
  * create some functions to be reused by default and rgb parsers
  * new commit fixing rgb option parser

### CODE REVIEW

File: src/bin/sol-fbp-generator/type-store.c [hunk 1]
```diff
@@ -321,11 +337,37 @@ read_default_value(struct decoder *d, struct option_description *o)
 }
 
 static bool
+check_types(struct sol_json_token *key,  struct sol_json_token *tmp, struct sol_json_token *value)
+{
+    if (sol_json_token_get_type(key) != SOL_JSON_TYPE_STRING)
```
Comment (lpereira) [inline src/bin/sol-fbp-generator/type-store.c]: No need to check this; the macro does that for you.

File: src/bin/sol-fbp-generator/type-store.c [hunk 2]
```diff
@@ -321,11 +337,37 @@ read_default_value(struct decoder *d, struct option_description *o)
...
+        return false;
+    if (sol_json_token_get_type(tmp) != SOL_JSON_TYPE_PAIR_SEP)
+        return false;
+    if (sol_json_token_get_type(value) != SOL_JSON_TYPE_NUMBER)
```
Comment (lpereira) [inline src/bin/sol-fbp-generator/type-store.c]: The value type, however, you still need to check.

**Comment by ibriano** [comment]
You need to add the Signed-off-by to every commit.

**Comment by bdilly** [comment]
Nice, changes done. Merged.

**Pull request closed without merging**